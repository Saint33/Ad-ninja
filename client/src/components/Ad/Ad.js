import React, { Component } from 'react';
import { Row, Col } from 'reactstrap';
import { Link } from 'react-router-dom';
import Spinner from 'react-spinkit';
import FaPhone from 'react-icons/lib/fa/phone';
import { formatDate, memberSince } from '../../utility';
import { connect } from 'react-redux';
import { getAd } from '../../actions/ad';

class Ad extends Component {

    componentWillMount(){
        this.props.dispatch(getAd(this.props.match.params.id))
            
    }

    render() {
        let ad = this.props.ad.currentAd;
        let loader = <Spinner className="loader-position" name='folding-cube' fadeIn="none"/>;
        return (
            <div>
                { !ad ? loader : 
            
            <Row className="adv-item">
            <Col xs={{ size: 7, offset: 1 }}>
                <div>
                    <h3 className="adv-item__title">{ad.title}</h3>
                    <span className="adv-item__number">№ 991446249, размещено {formatDate(ad.createdAt)}</span>
                </div>
                <img 
                    src={`/api/file/${ad.image}`}
                    className="adv-item__image"
                    />
                <div className="adv-item__info">
                    <div className="adv-item__info-address">
                        <span className="adv-item__info-address_title">Адрес:</span>
                        <span className="adv-item__info-address_value">{ad.address}</span>
                    </div>
                    <p className="adv-item__info-description"> 
                        {ad.description}
                    </p>

                </div>
            </Col>
            <Col className="adv-item__sideinfo" xs={{ size: 3 }}>
                <span className="adv-item__price">{ad.price} ₽</span>
                <div className="adv-item__selers-info">
                    <div className="adv-item__selers-info_phone">
                        <FaPhone size={20} className="phone-icon"/>
                        <span >{ad.owner.phone}</span>
                    </div>
                        <Link to={`/user/${ad.owner._id}`} className="adv-item__selers-info_name">{ad.owner.firstname}</Link>
                        <span className="adv-item__selers-info_reg">На Add-ninja c {memberSince(ad.owner.createdAt)}</span>
                    <div className="adv-item__selers-info_adress">
                        <span className="adv-item__selers-info_adress-title">Адрес:</span>
                        <span className="adv-item__selers-info_adress-value">{ad.owner.address}</span>

                    </div>
                </div>
            </Col>
            </Row>
            }
            </div>

        );
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        ad: state.ad
    }
}

export default connect(mapStateToProps)(Ad);