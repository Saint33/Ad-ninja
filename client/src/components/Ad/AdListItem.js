import React from 'react'
import { Link } from 'react-router-dom'
import { formatDate } from '../../utility'
import FaStar from 'react-icons/lib/fa/star'
import { connect } from 'react-redux'
import { addToFavorites } from '../../actions/ad'
import { openModal } from '../../actions/user'

const AdListItem = ({_id, image, title, price, createdAt, handleAddToFavorites, userId, className, VIP}) => (
    <div className={VIP ? "adv-list-item adv-list-item__vip" : "adv-list-item"}>
        <FaStar 
            size={26} 
            color="white" 
            className="adv-list-item__favorite" 
            onClick={() => handleAddToFavorites({id: userId, adId: _id})}
            />
        <Link to={`/ad/${_id}`}>
        <div className="adv-list-item__image-wrapper">
            <img 
                className="adv-list-item__image"
                alt="Обьявление" 
                src={`/api/file/${image}`}
            />
        </div>
        </Link>
        <div className="adv-list-item__description">
            <Link 
                to={`/ad/${_id}`} 
                className="adv-list-item__description_title">
                {title}
            </Link>
            <span 
                className="adv-list-item__description_price">
                {price} р.
            </span>
            <span 
                className="adv-list-item__description_date">
                {formatDate(createdAt)}
            </span>
        </div>
    </div>

);

const mapDispatchToProps = (dispatch) => {
    return {
        handleAddToFavorites: data => {
            if(data.id){
                dispatch(addToFavorites(data))
            } else {
                dispatch(openModal())
            }
        }
    }
}

export default connect(null, mapDispatchToProps)(AdListItem);