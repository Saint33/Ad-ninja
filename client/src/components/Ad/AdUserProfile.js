import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { deleteAd } from '../../actions/ad';
import { formatDate } from '../../utility'

const UserProfileAd = (ad) => (
    <div className="user-profile-ad">
        <div className="user-profile-ad__image-wrapper">
            <Link to={`/ad/${ad._id}`}>
                <img src={`/api/file/${ad.image}`} className="user-profile-ad__image"/>
            </Link>
        </div>
        <div className="user-profile-ad__content-wrapper">
            <h3 className="user-profile-ad__title">{ad.title}</h3>
            <span className="user-profile-ad__price">{ad.price}</span>
            <span className="user-profile-ad__date">Размещено {formatDate(ad.createdAt)}</span>
            <button className="user-profile-ad__button" onClick={() => ad.deleteCurrentAd(ad._id)}>Удалить</button>
        </div>
        
    </div>
)

const mapDispatchToProps = (dispatch) => {
    return {
        deleteCurrentAd: (id) => {
            dispatch(deleteAd(id))
        }
    }
}

export default connect(null, mapDispatchToProps)(UserProfileAd);