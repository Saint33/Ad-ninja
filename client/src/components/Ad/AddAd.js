import React from 'react';
import { Col,  Row, Form, FormGroup, Label, Input, FormText } from 'reactstrap';
import axios from 'axios';
import { connect } from 'react-redux';

class AddAd extends React.Component {
    state = {
        adInfo: {
            category: 'cars',
            title: '',
            description: '',
            price: '',
            address: ''
        },
        image: null
    }

    handleCategoryChange = (e) => {
        this.setState({adInfo:{...this.state.adInfo, category: e.target.value, }})
    }
    handleAddressChange = (e) => {
        this.setState({adInfo: {...this.state.adInfo, address: e.target.value}})
    }
    handleTitleChange = (e) => {
        this.setState({adInfo: {...this.state.adInfo, title: e.target.value}})
    }
    handleDescriptionChange = (e) => {
        this.setState({adInfo: {...this.state.adInfo, description: e.target.value}})
    }
    handlePriceChange = (e) => {
        this.setState({adInfo: {...this.state.adInfo, price: e.target.value}})
    }
    handleImageChange = (e) => {
        let file = e.target.files[0];
        let adImage = new FormData();
        adImage.append('file', file);
        this.setState({image: adImage})
    }
    handleFormSubmit = (e) => {
        e.preventDefault();
        axios.post('/api/ad', {...this.state.adInfo, ownerId: this.props.user.login.id}) 
            .then(response => {
                console.log(response);
                let id = response.data.AdId;
                let image = this.state.image;
                image.append('id', id);
                axios.post('/api/upload', image)
                    .then(response => console.log(response))
            })
    }

  render() {
      let user = this.props.user.login;
    return (
    <Row>
        
        <Col sm={8}>
        <h3 className="add-ad__title">Подать объявление</h3>
        <div className="contacts">
            <h4 className="contacts__title">Контактная информация</h4>
            <div>
                <div className="contacts__group">
                    <span className="contacts__item">Электронная почта </span>
                    <span className="contacts__value">{user.email}</span>
                </div>
                <div className="contacts__group">
                    <span className="contacts__item">Ваше имя</span>
                    <span className="contacts__value">{user.firstname}</span>
                </div>
                <div className="contacts__group">
                    <span className="contacts__item">Телефон</span>
                    <span className="contacts__value">{user.phone}</span>
                </div>
            </div>
        </div>
      <Form onSubmit={this.handleFormSubmit}>
        <FormGroup row>
          <Label for="exampleSelect" sm={2}>Категория</Label>
          <Col sm={5}>
            <Input type="select" name="category" onChange={this.handleCategoryChange}>
                <option value="cars">Авто</option>
                <option value="realestate">Недвижимость</option>
                <option value="jobs">Работа</option>
                <option value="electronics">Бытовая Электроника</option>
                <option value="personal">Личные вещи</option>
                <option value="hobby">Хобби</option>
            </Input>
          </Col>
        </FormGroup>
        <FormGroup row>
          <Label for="exampleEmail" sm={2}>Город</Label>
          <Col sm={10}>
            <Input type="text" name="address" onChange={this.handleAddressChange}/>
          </Col>
        </FormGroup>
        <FormGroup row>
          <Label for="exampleEmail" sm={2}>Название объявления</Label>
          <Col sm={10}>
            <Input type="text" name="title" onChange={this.handleTitleChange}/>
          </Col>
        </FormGroup>
        <FormGroup row>
          <Label for="exampleText" sm={2}>Описание объявления</Label>
          <Col sm={10}>
            <Input type="textarea" name="text" id="exampleText" onChange={this.handleDescriptionChange}/>
          </Col>
        </FormGroup>
        <FormGroup row>
          <Label for="exampleEmail" sm={2}>Цена</Label>
          <Col sm={10}>
            <Input type="number" name="price" onChange={this.handlePriceChange}/>
          </Col>
        </FormGroup>
        <FormGroup row>
          <Label for="exampleFile" sm={2}>Фотографии</Label>
          <Col sm={10}>
            <Input type="file" name="file" id="exampleFile" onChange={this.handleImageChange}/>
            <FormText color="muted">
              This is some placeholder block-level help text for the above input.
              It's a bit lighter and easily wraps to a new line.
            </FormText>
          </Col>
        </FormGroup>
        <button type="submit" className="header__button">Опубликовать</button>
      </Form>
      </Col>
      </Row>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
    return {
        user: state.user
    }
}

export default connect(mapStateToProps)(AddAd);