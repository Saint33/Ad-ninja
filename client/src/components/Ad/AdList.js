import React from 'react';
import AdListItem from './AdListItem';

const AdList = (props) => {
    
}

export default AdList;