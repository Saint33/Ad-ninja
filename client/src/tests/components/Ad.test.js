import React from 'react';
import { shallow } from 'enzyme';
import Ad from '../../components/Ad/Ad';


describe('Ad', () => {
    const tree = shallow(
        <Ad 
            match={{params: {id: '5ae451e95c66692bb8b34e50'}}}
        />
    );
    it('should render correctly',  () => {
        expect(tree).toMatchSnapshot();
    })
})