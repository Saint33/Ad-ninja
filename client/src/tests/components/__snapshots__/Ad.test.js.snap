// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ad should render correctly 1`] = `
<div>
  <Loader
    active={true}
    innerClassName="loader-position"
    type="ball-clip-rotate-multiple"
  />
</div>
`;
